macros/attribution_overview.sql

```diff
@@ -7,7 +7,66 @@
 
 {%- set __, last_processed_cv_tstamp = snowplow_utils.return_limits_from_model(ref('snowplow_attribution_campaign_attributions'),'cv_tstamp','cv_tstamp') %}
 
-with campaign_prep as (
+-- making sure we only include spend that is needed
+with campaign_spend as (
+
+  {% if var('snowplow__spend_source') != 'not defined' %}
+    select distinct s.campaign, s.spend, s.spend_tstamp
+    from {{ var('snowplow__spend_source') }} s
+    inner join {{ ref('snowplow_attribution_campaign_attributions') }} c
+    on c.campaign = s.campaign and s.spend_tstamp < cv_tstamp 
+    and s.spend_tstamp > {{ snowplow_utils.timestamp_add('day', -90, 'cv_tstamp') }}
+  
+  {% else %}
+    select true
+  {% endif %}
+    
+)
+
+, channel_spend as (
+  
+  {% if var('snowplow__spend_source') != 'not defined' %}
+    select distinct s.channel,  s.spend, s.spend_tstamp 
+    from {{ var('snowplow__spend_source') }} s
+    inner join {{ ref('snowplow_attribution_channel_attributions') }} c
+    on c.channel = s.channel and s.spend_tstamp < cv_tstamp
+    and s.spend_tstamp > {{ snowplow_utils.timestamp_add('day', -90, 'cv_tstamp') }}
+  
+  {% else %}
+    select true
+  {% endif %}
+    
+)
+
+-- grouping spend to avoid duplicates in later join
+, campaign_spend_grouped as (
+
+  {% if var('snowplow__spend_source') != 'not defined' %}
+    select campaign, sum(spend) as spend
+    from campaign_spend
+    group by 1
+  
+  {% else %}
+    select true
+  {% endif %}
+    
+)
```

Comment:
Should we not do a sum in the first one as well? If they have a 1:many channel:campaign table we would want to sum there, no?

Reply:
It will be summed in the subsequent cte and we do not lose the data because then the campaign will be different. We will only lose exact duplicates but we can't be outsmart badly created source data. What I forgot to do though, is to filter out null channels/campaigns. Or did I miss the scenario you had in mind?
<img width="309" alt="image" src="https://github.com/snowplow/dbt-snowplow-attribution/assets/95634439/1fac1a4d-28a7-4147-b500-dee0d883ece5">

Reply:
We need to do the deduping as there can be duplicate lines due to the join as there can be multiple conversions per spend.. I added a spend_id key to preserve everything in spend while being able to dedupe.
<img width="1194" alt="image" src="https://github.com/snowplow/dbt-snowplow-attribution/assets/95634439/00c84f4b-7334-424f-a93f-8b8819efd302">

Reply:
Looks good, are we also going to make clear in the docs what format we expect this to be in now?

Reply:
yeah I will put something in, shall I add the changes to your unified docs pr? we might have some rebase conflicts otherwise

Reply:
I don't think I touched the attribution stuff so you should be fine to do a different PR?